tox.ini

```diff
@@ -0,0 +1,45 @@
+[tox]
+requires =
+    tox>=4
+envlist =
+    # Test using the system Python.
+    test,
+    # Run pre-commit on all files.
+    lint,
+    # Run pre-commit on all files, including stages that require manual fixes.
+    lint-manual,
+    # Typecheck all files.
+    typecheck
+
+[testenv:test]
+description = run unit tests
+commands =
+    python --version
+    python setup.py test {posargs}
+
+[testenv:lint]
+description = run pre-commit
+deps =
+    pre-commit
+commands =
+    pre-commit run --all-files
+
+[testenv:lint-manual]
+description = run all pre-commit stages, including those that require manual fixes
+deps =
+    pre-commit
+commands =
+    pre-commit run --all-files
+    pre-commit run --all-files --hook-stage manual flake8
+    pre-commit run --all-files --hook-stage manual doc8
```

Comment:
`pre-commit run --all-files --hook-stage manual` will run all the pre-commit checks in one command.

Reply:
Good catch, thanks!